File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> map = new HashMap<>();
        for (String person : participant) {
            map.put(person, map.getOrDefault(person,0)+1);
        }
        for (String c : completion) {
            int n = map.get(c) - 1;
            if (n > 0) map.put(c,n);
            else map.remove(c); 
        }
        return map.keySet().iterator().next();
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    unordered_map<string, int> mp;
    
    for (string name: completion) {
        mp[name]++;
    }
    
    for (string name: participant) {
        if (mp[name] == 0) {
            return name;
        }
        mp[name]--;
    }
    return "";
}